In [1]:
Analyses des causes des accidents

SyntaxError: invalid syntax (656561196.py, line 1)

In [5]:
import pandas as pd
import numpy as np

In [6]:
df_caract = pd.read_csv("Data/caract-2023.csv", encoding = 'utf-8', delimiter = ";")
df_lieux = pd.read_csv("Data/lieux-2023.csv", encoding = 'utf-8', delimiter = ";")
df_usagers = pd.read_csv("Data/usagers-2023.csv", encoding = 'utf-8', delimiter = ";")
df_vehicules = pd.read_csv("Data/vehicules-2023.csv", encoding = 'utf-8', delimiter = ";")

/tmp/ipykernel_1564/2696134291.py:2: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  df_lieux = pd.read_csv("Data/lieux-2023.csv", encoding = 'utf-8', delimiter = ";")


In [ ]:
Etude du fichier lieux

In [7]:
df_lieux.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70860 entries, 0 to 70859
Data columns (total 18 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Num_Acc  70860 non-null  int64 
 1   catr     70860 non-null  int64 
 2   voie     58113 non-null  object
 3   v1       70860 non-null  int64 
 4   v2       5884 non-null   object
 5   circ     70860 non-null  int64 
 6   nbv      70860 non-null  object
 7   vosp     70860 non-null  int64 
 8   prof     70860 non-null  int64 
 9   pr       70860 non-null  object
 10  pr1      70860 non-null  object
 11  plan     70860 non-null  int64 
 12  lartpc   31 non-null     object
 13  larrout  70860 non-null  object
 14  surf     70860 non-null  int64 
 15  infra    70860 non-null  int64 
 16  situ     70860 non-null  int64 
 17  vma      70860 non-null  int64 
dtypes: int64(11), object(7)
memory usage: 9.7+ MB


In [8]:
df_lieux.isnull().sum(axis =0)

Num_Acc        0
catr           0
voie       12747
v1             0
v2         64976
circ           0
nbv            0
vosp           0
prof           0
pr             0
pr1            0
plan           0
lartpc     70829
larrout        0
surf           0
infra          0
situ           0
vma            0
dtype: int64

In [9]:
df_lieux.describe()

Num_Acc          catr            v1          circ          vosp  \
count  7.086000e+04  70860.000000  70860.000000  70860.000000  70860.000000   
mean   2.023000e+11      3.397403     -0.228860      1.712673      0.183164   
std    1.581802e+04      1.113553      0.431536      0.942601      0.744139   
min    2.023000e+11      1.000000     -1.000000     -1.000000     -1.000000   
25%    2.023000e+11      3.000000      0.000000      1.000000      0.000000   
50%    2.023000e+11      3.000000      0.000000      2.000000      0.000000   
75%    2.023000e+11      4.000000      0.000000      2.000000      0.000000   
max    2.023001e+11      9.000000      3.000000      4.000000      3.000000   

               prof         plan          surf         infra          situ  \
count  70860.000000  70860.00000  70860.000000  70860.000000  70860.000000   
mean       1.206336      1.26733      1.254036      0.811177      1.611727   
std        0.534770      0.65518      0.756575      2.162585      1.627796   
min       -1.000000     -1.00000     -1.000000     -1.000000     -1.000000   
25%        1.000000      1.00000      1.000000      0.000000      1.000000   
50%        1.000000      1.00000      1.000000      0.000000      1.000000   
75%        1.000000      1.00000      1.000000      0.000000      1.000000   
max        4.000000      4.00000      9.000000      9.000000      8.000000   

                vma  
count  70860.000000  
mean      53.306393  
std       26.001154  
min       -1.000000  
25%       30.000000  
50%       50.000000  
75%       70.000000  
max      130.000000

In [10]:
df_lieux['catr'].value_counts(normalize =True)

catr
4    0.456732
3    0.377251
1    0.073130
2    0.056661
7    0.026136
6    0.005828
9    0.003260
5    0.001002
Name: proportion, dtype: float64

In [11]:
df_lieux['circ'].value_counts(normalize =True)

circ
 2    0.613647
 1    0.198095
 3    0.111459
-1    0.070858
 4    0.005941
Name: proportion, dtype: float64

In [12]:
print(len(df_lieux['nbv'].unique()))

15


In [13]:
df_lieux['nbv'].value_counts(normalize =True)

nbv
2               0.600903
1               0.118445
4               0.085972
3               0.072043
 -1             0.058609
0               0.029354
6               0.016892
5               0.010246
8               0.004121
7               0.001327
#VALEURMULTI    0.000762
10              0.000677
9               0.000325
12              0.000226
11              0.000099
Name: proportion, dtype: float64

In [14]:
df_lieux['vosp'].value_counts(normalize =True)

vosp
 0    0.815072
-1    0.055800
 1    0.055715
 2    0.036988
 3    0.036424
Name: proportion, dtype: float64

In [15]:
df_lieux['prof'].value_counts(normalize =True)

prof
 1    0.824287
 2    0.145795
 3    0.014324
 4    0.012616
-1    0.002978
Name: proportion, dtype: float64

In [16]:
df_lieux['pr'].value_counts(normalize =True)

pr
 -1      0.420491
0        0.202272
1        0.041575
2        0.020519
3        0.017866
           ...   
443      0.000014
510      0.000014
390      0.000014
1 630    0.000014
368      0.000014
Name: proportion, Length: 446, dtype: float64

In [17]:
df_lieux['pr1'].value_counts(normalize =True)

pr1
 -1      0.423144
0        0.208524
1        0.031471
500      0.026447
200      0.014776
           ...   
1 023    0.000014
1 139    0.000014
752      0.000014
1 236    0.000014
562      0.000014
Name: proportion, Length: 1363, dtype: float64

In [18]:
df_lieux['plan'].value_counts(normalize =True)

plan
 1    0.824640
 2    0.084970
 3    0.077237
 4    0.010838
-1    0.002314
Name: proportion, dtype: float64

In [19]:
df_lieux['larrout'].value_counts(normalize =True)

larrout
 -1     0.742083
7       0.062221
5,5     0.045809
4       0.030003
5       0.028422
          ...   
47      0.000014
12,7    0.000014
2,1     0.000014
6,65    0.000014
2,85    0.000014
Name: proportion, Length: 107, dtype: float64

In [20]:
df_lieux['surf'].value_counts(normalize =True)

surf
 1    0.789867
 2    0.196585
 9    0.004177
-1    0.002935
 7    0.001722
 3    0.001609
 8    0.001185
 5    0.000917
 6    0.000692
 4    0.000310
Name: proportion, dtype: float64

In [21]:
df_lieux['infra'].value_counts(normalize =True)

infra
 0    0.833841
 5    0.066836
 9    0.032134
 2    0.014832
-1    0.013026
 6    0.010020
 3    0.009780
 1    0.008877
 8    0.007084
 4    0.002949
 7    0.000621
Name: proportion, dtype: float64

In [22]:
df_lieux['situ'].value_counts(normalize =True)

situ
 1    0.836283
 3    0.058651
 8    0.037863
 5    0.027293
 4    0.020773
 6    0.011784
 2    0.004558
-1    0.002794
Name: proportion, dtype: float64

In [23]:
df_lieux['vma'].value_counts(normalize =True)

vma
 50     0.453387
 30     0.184505
 80     0.117570
 90     0.068233
-1      0.064606
 70     0.060076
 110    0.028662
 130    0.015368
 20     0.003373
 60     0.001708
 25     0.000974
 40     0.000917
 45     0.000212
 100    0.000169
 10     0.000085
 15     0.000056
 75     0.000028
 35     0.000028
 31     0.000014
 65     0.000014
 23     0.000014
Name: proportion, dtype: float64

<!-- # Commentaires fichier lieux
# 70860 entrées, 17 colonnes, dont 11 types entiers et 7 types objets
# 3 colonnes contiennent des valeurs nulles, la colonnes VOIE indique le numéro de la route, la colonne V2 indique la lettre indice alphanumérique de la voie, et la colonne lartpc(largeur du terre plein central)n'est pas renseignée. on pourra supprimer ces 3 colonnes
# catr : catégorie de route: les accidents ont lieux sur les voies communales (45% ) et les routes départementales (38%)
# Voie :la colonnes VOIE indique le numéro de la route, trop de valeurs nulles  A SUPPRIMER
# V1 : indice numerique de la route (bis, ter..., A SUPPRIMER
# V2 : supprimer trop de velaurs nulles A SUPPRIMER
# circ : régime de circulation : 61 % des accidents ont lieux sur des routes bidrectionnelles, 20% sur des routes  à sens unique (7% de valeurs non renseignées)
# nbv : nombre total de voies de circulation: 60% 2 voies de circulation (5% non renseigné)
# vosp : signale l'existence d'une voie réservée, indépendemment du fait que l'accident ait lieu ou non sur cette voie (80% sans objet) A SUPPRIMER
# prof : Profil en long décrit la déclivité de la route : 82 % profil plat( -1 % non renseigner)
# pr : Numero de PR de rattachement (numero de la borne amont): pas d'intérêt sur les causes et non renseigné (40%): A SUPPRIMER
# pr1 : distance en m de la borne de rattachement:pas d'intérêt sur les causes et non renseigné (40%): A SUPPRIMER
# plan:82% en partie rectiligne( -1 % non renseigner)
# lartpc: non renseigner: A SUPPRIMER
# larrout : 75 % des données non renseignées A SUPPRIMER
# surf :  etat de surface: 80 % des cas  dus  à une surface normale ( -1 % non renseigner)
# infra:  Aménagement-Infrastructure : 80% des cas pas d'ammenagement( 1,3 % non renseigner)
# situ :  situation de l'accident: 80% sur chaussée ( -1 % non renseigner)
# vma: vitesse maximale autorisée sur le lieu au moment de l'accident: dans 45% des cas  50 km/h et ( -1 % non renseigner) -->


Commentaires Fichier lieux<br/>
70860 entrées, 17 colonnes, dont 11 types entiers et 7 types objets<br/>
3 colonnes contiennent des valeurs nulles, la colonnes VOIE indique le numéro de la route, la colonne V2 indique la lettre indice alphanumérique de la voie, et la colonne lartpc(largeur du terre plein central)n'est pas renseignée. on pourra supprimer ces 3 colonnes<br/>
catr : catégorie de route: les accidents ont lieux sur les voies communales (45% ) et les routes départementales (38%)<br/>
Voie :la colonnes VOIE indique le numéro de la route, trop de valeurs nulles  A SUPPRIMER<br/>
V1 : indice numerique de la route (bis, ter..., A SUPPRIMER<br/>
V2 : supprimer trop de velaurs nulles A SUPPRIMER<br/>
circ : régime de circulation : 61 % des accidents ont lieux sur des routes bidrectionnelles, 20% sur des routes  à sens unique (7% de valeurs non renseignées)<br/>
nbv : nombre total de voies de circulation: 60% 2 voies de circulation (5% non renseigné)<br/>
vosp : signale l'existence d'une voie réservée, indépendemment du fait que l'accident ait lieu ou non sur cette voie (80% sans objet) A SUPPRIMER<br/>
prof : Profil en long décrit la déclivité de la route : 82 % profil plat( -1 % non renseigner)<br/>
pr : Numero de PR de rattachement (numero de la borne amont): pas d'intérêt sur les causes et non renseigné (40%): A SUPPRIMER<br/>
pr1 : distance en m de la borne de rattachement:pas d'intérêt sur les causes et non renseigné (40%): A SUPPRIMER<br/>
plan:82% en partie rectiligne( -1 % non renseigner)<br/>
lartpc: non renseigner: A SUPPRIMER<br/>
larrout : 75 % des données non renseignées A SUPPRIMER<br/>
surf :  etat de surface: 80 % des cas  dus  à une surface normale ( -1 % non renseigner)<br/>
infra:  Aménagement-Infrastructure : 80% des cas pas d'ammenagement( 1,3 % non renseigner)<br/>
situ :  situation de l'accident: 80% sur chaussée ( -1 % non renseigner)<br/>
vma: vitesse maximale autorisée sur le lieu au moment de l'accident: dans 45% des cas  50 km/h et ( -1 % non renseigner)<br/>

In [ ]:
# fichier usager

In [24]:
df_usagers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125789 entries, 0 to 125788
Data columns (total 16 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Num_Acc      125789 non-null  int64  
 1   id_usager    125789 non-null  object 
 2   id_vehicule  125789 non-null  object 
 3   num_veh      125789 non-null  object 
 4   place        125789 non-null  int64  
 5   catu         125789 non-null  int64  
 6   grav         125789 non-null  int64  
 7   sexe         125789 non-null  int64  
 8   an_nais      123191 non-null  float64
 9   trajet       125789 non-null  int64  
 10  secu1        125789 non-null  int64  
 11  secu2        125789 non-null  int64  
 12  secu3        125789 non-null  int64  
 13  locp         125789 non-null  int64  
 14  actp         125789 non-null  object 
 15  etatp        125789 non-null  int64  
dtypes: float64(1), int64(11), object(4)
memory usage: 15.4+ MB


In [25]:
df_usagers.isnull().sum(axis =0)

Num_Acc           0
id_usager         0
id_vehicule       0
num_veh           0
place             0
catu              0
grav              0
sexe              0
an_nais        2598
trajet            0
secu1             0
secu2             0
secu3             0
locp              0
actp              0
etatp             0
dtype: int64

In [26]:
df_usagers.describe()

Num_Acc          place           catu           grav  \
count  1.257890e+05  125789.000000  125789.000000  125789.000000   
mean   2.023000e+11       2.091646       1.332684       2.514544   
std    1.582901e+04       2.582452       0.611056       1.377953   
min    2.023000e+11      -1.000000       1.000000      -1.000000   
25%    2.023000e+11       1.000000       1.000000       1.000000   
50%    2.023000e+11       1.000000       1.000000       3.000000   
75%    2.023000e+11       2.000000       2.000000       4.000000   
max    2.023001e+11      10.000000       3.000000       4.000000   

                sexe        an_nais         trajet          secu1  \
count  125789.000000  123191.000000  125789.000000  125789.000000   
mean        1.274134    1984.252023       3.128692       1.932323   
std         0.561198      19.099389       2.754842       2.371647   
min        -1.000000    1913.000000      -1.000000      -1.000000   
25%         1.000000    1971.000000       0.000000       1.000000   
50%         1.000000    1988.000000       4.000000       1.000000   
75%         2.000000    2000.000000       5.000000       2.000000   
max         2.000000    2023.000000       9.000000       9.000000   

               secu2          secu3           locp          etatp  
count  125789.000000  125789.000000  125789.000000  125789.000000  
mean        0.902885      -0.885284      -0.236046      -0.816232  
std         2.980147       1.004342       1.279765       0.640256  
min        -1.000000      -1.000000      -1.000000      -1.000000  
25%        -1.000000      -1.000000      -1.000000      -1.000000  
50%         0.000000      -1.000000      -1.000000      -1.000000  
75%         0.000000      -1.000000       0.000000      -1.000000  
max         9.000000       9.000000       9.000000       3.000000

In [27]:
df_usagers['place'].value_counts(normalize =True)

place
 1     0.742831
 2     0.111774
 10    0.075690
 3     0.019239
 4     0.018054
 9     0.010398
 7     0.009683
 5     0.006074
 8     0.004635
 6     0.001606
-1     0.000016
Name: proportion, dtype: float64

In [28]:
df_usagers['catu'].value_counts(normalize =True)

catu
1    0.743006
2    0.181304
3    0.075690
Name: proportion, dtype: float64

In [29]:
df_usagers['grav'].value_counts(normalize =True)

grav
 1    0.424512
 4    0.394335
 3    0.153201
 2    0.027013
-1    0.000938
Name: proportion, dtype: float64

In [30]:
df_usagers['sexe'].value_counts(normalize =True)

sexe
 1    0.667888
 2    0.312786
-1    0.019326
Name: proportion, dtype: float64

In [31]:
df_usagers['an_nais'].value_counts(normalize =True)

an_nais
2003.0    0.030083
2001.0    0.029182
2002.0    0.028939
2000.0    0.028785
2004.0    0.026909
            ...   
1921.0    0.000016
1917.0    0.000008
1920.0    0.000008
1919.0    0.000008
1913.0    0.000008
Name: proportion, Length: 107, dtype: float64

In [32]:
df_usagers['trajet'].value_counts(normalize =True)

trajet
 5    0.366733
 0    0.264753
 1    0.130973
 4    0.087472
 9    0.077861
 3    0.028603
 2    0.023738
-1    0.019867
Name: proportion, dtype: float64

In [33]:
df_usagers['secu1'].value_counts(normalize =True)

secu1
 1    0.580242
 2    0.174991
 8    0.123063
 0    0.094905
-1    0.018221
 3    0.005772
 6    0.001073
 9    0.000978
 4    0.000533
 5    0.000215
 7    0.000008
Name: proportion, dtype: float64

In [34]:
df_usagers['secu2'].value_counts(normalize =True)

secu2
-1    0.440571
 0    0.354856
 6    0.094539
 8    0.078966
 4    0.012123
 5    0.009174
 9    0.003713
 1    0.002456
 3    0.001455
 7    0.001121
 2    0.001026
Name: proportion, dtype: float64

In [35]:
df_usagers['secu3'].value_counts(normalize =True)

secu3
-1    0.980944
 9    0.008133
 0    0.006241
 8    0.001868
 1    0.001590
 6    0.000636
 2    0.000191
 4    0.000175
 5    0.000135
 3    0.000064
 7    0.000024
Name: proportion, dtype: float64

In [36]:
df_usagers['locp'].value_counts(normalize =True)

locp
-1    0.503462
 0    0.416205
 3    0.026004
 2    0.015725
 4    0.012155
 1    0.011567
 5    0.006694
 9    0.004221
 6    0.002671
 8    0.001208
 7    0.000087
Name: proportion, dtype: float64

In [37]:
df_usagers['actp'].value_counts(normalize =True)

actp
 -1    0.502850
0      0.416849
3      0.059186
9      0.005517
1      0.005215
B      0.002973
5      0.002751
2      0.002512
A      0.000938
4      0.000715
6      0.000231
7      0.000191
8      0.000072
Name: proportion, dtype: float64

In [39]:
df_usagers['etatp'].value_counts(normalize =True)

etatp
-1    0.919715
 1    0.060371
 2    0.016631
 3    0.003283
Name: proportion, dtype: float64

Commentaires Fichier usager 

tableau 125000 lignes , 16 colonnes: 4 types objet, 1float, 11 types entiers

Num_ACC: à conserver
id_usager  : A SUPPRIMER
id_vehicule : A SUPPRIMER
num_Veh : A SUPPRIMER
place : Essentiellement en place conducteur(74%) et passager avant(11%) puis piéton (7%)
catu  : Même information que place (en supprimer 1 des 2)
grav :TARGET
sexe : 66% d'hommes et 32% de femmes
An_nais : année de naissance de l'usager:manque des valeurs et plus de 107 critères, à regrouper par tranche ?
trajet: 36,6 % pour les loisirs mais 26% non renseignés?
secu1: équipement de sécurité de l'usager: 58 % porte 1 ceinture,17% porte 1 casque et 12 % ce n'est pas identifiable
secu2: équipement de sécurité de l'usager: pas renseigné (44%)ou pas de second équipement (35%)
secu3: non renseigné à 98 % A SUPPRIMER
locp : localisation du pieton 91% non renseigneé+ sans objet  A SUPPRIMER
actp: action du piéton (95% non renseigné  ou sans objet)A SUPPRIMER
etap : piéton accidenté  accompagné ou non( non renseigné à 91% ) A SUPPRIMER











